### PR TITLE
[zh-cn]: update the translation of var in statement section

### PR DESCRIPTION
**Pull request by JasonLamv-t**
### Description

As the title.

### Motivation

Ditto.

### Additional details

None.

### Related issues and pull requests

#18266

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/statements/var/index.md [hunk 1]
```diff
@@ -1,142 +1,186 @@
...
 title: var
 slug: Web/JavaScript/Reference/Statements/var
+l10n:
+  sourceCommit: 4f86aad2b0b66c0d2041354ec81400c574ab56ca
 ---
 
 {{jsSidebar("Statements")}}
 
-**`var` 语句** 用于声明一个函数范围或全局范围的变量，并可将其初始化为一个值（可选）。
+**`var` 语句** 用于声明一个函数作用域或全局作用域的变量，并且可以选择将其初始化为一个值。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/statements/var/index.md:10]:
```suggestion
**`var` 语句**用于声明一个函数作用域或全局作用域的变量，并且可以选择将其初始化为一个值。
```

File: files/zh-cn/web/javascript/reference/statements/var/index.md [hunk 2]
```diff
@@ -1,142 +1,186 @@
...
-## 描述
+- 函数主体
+- 类[静态初始化块](/zh-CN/docs/Web/JavaScript/Reference/Classes/Static_initialization_blocks)
 
-无论在何处声明变量，都会在执行任何代码之前进行处理。这被称为{{Glossary("Hoisting", "变量提升")}}，我们将在下面进一步讨论。
+如果不是以上这些情况则是：
 
-用 `var` 声明的变量的作用域是它当前的执行上下文及其闭包（嵌套函数），或者对于声明在任何函数外的变量来说是全局。使用 `var` 重复声明 JavaScript 变量并不会抛出错误（即使在严格模式 (strict mode) 下），同时，变量也不会丢失其值，直到调用其他的赋值操作。
+- 当前[模块](/zh-CN/docs/Web/JavaScript/Guide/Modules)，如果代码以模块模式运行
+- 全局作用域，如果代码以脚本模式运行。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/statements/var/index.md:39]:
```suggestion
- 全局作用域，如果代码以脚本模式运行
```

File: files/zh-cn/web/javascript/reference/statements/var/index.md [hunk 3]
```diff
@@ -1,142 +1,186 @@
...
+Object.hasOwn(globalThis, "x"); // true
+delete globalThis.x; // 在严格模式下，将抛出 TypeError，否则静默失败。
 delete x; // 在严格模式下，将抛出 SyntaxError，否则静默失败。
 ```
 
-注意，在 Node.js [CommonJS](http://www.commonjs.org/) modules 以及原生 [ECMAScript modules](/zh-CN/docs/Web/JavaScript/Guide/Modules) 中，顶层变量被声明在模块（module）的作用域内，而不是以属性的形式被添加到全局对象中。
+在 NodeJS [CommonJS](http://www.commonjs.org/) 模块以及原生 [ECMAScript 模块](/zh-CN/docs/Web/JavaScript/Guide/Modules)中，顶层变量声明的作用域仅限于模块中，而不会作为属性被添加到全局对象中。
 
-### 未限定标识符的赋值
+`var` 关键字后面的列表被称为*{{Glossary("binding", "绑定")}}列表*，用逗号分隔，其中逗号*不是*[逗号运算符](/zh-CN/docs/Web/JavaScript/Reference/Operators/Comma_operator)，`=` 号也*不是*[赋值运算符](/zh-CN/docs/Web/JavaScript/Reference/Operators/Assignment)。后续变量的初始化可以引用前面的变量，并获得初始化的值。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/statements/var/index.md:76]:
```suggestion
`var` 关键字后面的列表被称为_{{Glossary("binding", "绑定")}}列表_，用逗号分隔，其中逗号*不是*[逗号运算符](/zh-CN/docs/Web/JavaScript/Reference/Operators/Comma_operator)，`=` 号也*不是*[赋值运算符](/zh-CN/docs/Web/JavaScript/Reference/Operators/Assignment)。后续变量的初始化可以引用前面的变量，并获得初始化的值。
```

File: files/zh-cn/web/javascript/reference/statements/var/index.md [hunk 4]
```diff
@@ -1,142 +1,186 @@
...
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/var/index.md:76]: 这样没法正常渲染斜体，

```suggestion
`var` 关键字后面的列表被称为[_绑定_](/zh-CN/docs/Glossary/Binding)*列表*，用逗号分隔，其中逗号*不是*[逗号运算符](/zh-CN/docs/Web/JavaScript/Reference/Operators/Comma_operator)，`=` 号也*不是*[赋值运算符](/zh-CN/docs/Web/JavaScript/Reference/Operators/Assignment)。后续变量的初始化可以引用前面的变量，并获得初始化的值。
```

File: files/zh-cn/web/javascript/reference/statements/var/index.md [hunk 5]
```diff
@@ -1,142 +1,186 @@
...
     var y = 2;
-    console.log(x); // 1 (函数 `bar` 包含了 `x`)
-    console.log(y); // 2 (`y` 在作用域内)
+    console.log(x); // 1（`bar` 函数闭包中引用了 `x`）
+    console.log(y); // 2（`y` 在作用域内）
   }
   bar();
-  console.log(x); // 1 (`x` 在作用域内)
-  console.log(y); // 在严格模式（strict mode）下将抛出 ReferenceError，`y` 仅在 `bar` 函数的作用域内
+  console.log(x); // 1（`x` 在范围内）
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/var/index.md:50]:
```suggestion
  console.log(x); // 1（`x` 在作用域内）
```

File: files/zh-cn/web/javascript/reference/statements/var/index.md [hunk 6]
```diff
@@ -1,142 +1,186 @@
...
+  console.log(y); // ReferenceError，`y` 的作用域限定在 `bar` 内
 }
 
 foo();
 ```
 
-使用 `var` 声明的变量将在任何代码执行前被创建，这被称为变量提升。这些变量的初始值为 `undefined`。
+重要的是，其他块级结构，包括[块语句](/zh-CN/docs/Web/JavaScript/Reference/Statements/block)、{{jsxref("Statements/try...catch", "try...catch")}}、{{jsxref("Statements/switch", "switch")}}以及[其中一个 `for` 语句的头部](/zh-CN/docs/Web/JavaScript/Reference/Statements#迭代)，对于 `var` 并不创建作用域，而在这样的块内部使用 `var` 声明的变量仍然可以在块外部被引用。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/var/index.md:57]:
```suggestion
重要的是，其他块级结构，包括[块语句](/zh-CN/docs/Web/JavaScript/Reference/Statements/block)、{{jsxref("Statements/try...catch", "try...catch")}}、{{jsxref("Statements/switch", "switch")}} 以及[其中一个 `for` 语句的头部](/zh-CN/docs/Web/JavaScript/Reference/Statements#迭代)，对于 `var` 并不创建作用域，而在这样的块内部使用 `var` 声明的变量仍然可以在块外部被引用。
```

File: files/zh-cn/web/javascript/reference/statements/var/index.md [hunk 7]
```diff
@@ -1,142 +1,186 @@
...
-console.log("still going..."); // still going...
-var x = 1;
-console.log(x); // 1
-console.log("still going..."); // still going...
+for (var a of [1, 2, 3]);
+console.log(a); // 3
 ```
 
-在全局上下文中，使用 `var` 声明的变量将作为全局对象的**不可配置**属性被添加。这意味着它的属性描述符无法被修改，也无法使用 {{JSxRef("Operators/delete", "delete")}} 删除。其对应的名字也将被添加到 [全局环境记录（global environment record）](https://www.ecma-international.org/ecma-262/10.0/index.html#sec-global-environment-records)（它构成了全局词法环境 (global lexical environment) 的一部分）的 `[[VarNames]]` 插槽内的列表中。`[[VarNames]]` 中的命名列表使运行时能够区分“全局变量”和“全局对象的直接属性”。
+在脚本中，使用 `var` 声明的变量将被添加为全局对象的**不可配置**属性。这意味着它的属性描述符无法被修改，也无法使用 {{JSxRef("Operators/delete", "delete")}} 删除。JavaScript 具有自动内存管理机制，因此在全局变量上使用 `delete` 运算符是没有意义的。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/var/index.md:64]:
```suggestion
在脚本中，使用 `var` 声明的变量将被添加为全局对象的不可配置属性。这意味着它的属性描述符无法被修改，也无法使用 {{JSxRef("Operators/delete", "delete")}} 删除。JavaScript 具有自动内存管理机制，因此在全局变量上使用 `delete` 运算符是没有意义的。
```

File: files/zh-cn/web/javascript/reference/statements/var/index.md [hunk 8]
```diff
@@ -1,142 +1,186 @@
...
+在 NodeJS [CommonJS](http://www.commonjs.org/) 模块以及原生 [ECMAScript 模块](/zh-CN/docs/Web/JavaScript/Guide/Modules)中，顶层变量声明的作用域仅限于模块中，而不会作为属性被添加到全局对象中。
 
-### 未限定标识符的赋值
+`var` 关键字后面的列表被称为[_绑定_](/zh-CN/docs/Glossary/Binding)_列表_，用逗号分隔，其中逗号*不是*[逗号运算符](/zh-CN/docs/Web/JavaScript/Reference/Operators/Comma_operator)，`=` 号也*不是*[赋值运算符](/zh-CN/docs/Web/JavaScript/Reference/Operators/Assignment)。后续变量的初始化可以引用前面的变量，并获得初始化的值。
 
-全局对象位于作用域链的顶部。在尝试将名称解析为值时，将沿着作用域链搜索该名称。这意味着，全局对象中的属性在所有作用域中均可见，而无需使用 `globalThis.`、`window.` 或 `global.` 来限定标识符。
+### 提升
 
-所以你可以这样写：
+`var` 声明，无论它们出现在脚本中的什么位置，都会在执行脚本中的任何代码之前进行处理。在代码中的任何位置声明变量都相当于在顶部声明它。这也意味着变量可以在其声明之前被使用。这种行为被称为[_提升_](/zh-CN/docs/Glossary/Hoisting)，因为变量声明似乎被移动到发生该行为的函数、静态初始化块或脚本源的顶部。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/var/index.md:80]:
```suggestion
`var` 声明，无论它们出现在脚本中的什么位置，都会在执行脚本中的任何代码之前进行处理。在代码中的任何位置声明变量都相当于在顶部声明它。这也意味着变量可以在其声明之前被使用。这种行为被称为[_提升_](/zh-CN/docs/Glossary/Hoisting)，因为变量声明似乎被移动到发生该行为的函数、静态初始化块或脚本源代码的顶部。
```

File: files/zh-cn/web/javascript/reference/statements/var/index.md [hunk 9]
```diff
@@ -1,142 +1,186 @@
...
 
-var bla;
-bla = 2;
+```js
+var a = 1;
+function a() {}
+console.log(a); // 1
+```
+
+`var` 不能与 `let`、`const`、`class` 或 `import` 在同一作用域中声明同名变量。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/var/index.md:139]:
```suggestion
`var` 不能与 {{jsxref("Statements/let", "let")}}、{{jsxref("Statements/const", "const")}}、{{jsxref("Statements/class", "class")}} 或 {{jsxref("Statements/import", "import")}} 在同一作用域中声明同名变量。
```

File: files/zh-cn/web/javascript/reference/statements/var/index.md [hunk 10]
```diff
@@ -1,142 +1,186 @@
...
-  console.log(bar); // undefined
-  bar = 111;
-  console.log(bar); // 111
+`catch` 块内的 `var` 声明可以与 `catch` 绑定的标识符同名，但只有当 `catch` 绑定的是一个简单标识符，而不是解构模式时才可以。这是一种[已弃用的语法](/zh-CN/docs/Web/JavaScript/Reference/Deprecated_and_obsolete_features#语句)，你不应该依赖它。在这种情况下，声明会被提升到 `catch` 块外部，但在 `catch` 块内的任何赋值都不会在外部可见。
+
+```js-nolint example-bad
+try {
+  throw 1;
+} catch (e) {
+  var e = 2; // Works
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/var/index.md:181]:
```suggestion
  var e = 2; // 有效
```

File: files/zh-cn/web/javascript/reference/statements/var/index.md [hunk 11]
```diff
@@ -202,12 +249,12 @@ console.log(x, y);
 
 ### 隐式全局变量和外部函数作用域
 
-看起来像是隐式全局作用域的变量也有可能是其外部函数变量的引用：
+看似隐式全局变量的变量可能是对外部函数作用域中变量的引用：
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/var/index.md:252]: 感觉原翻译更好一些

File: files/zh-cn/web/javascript/reference/statements/var/index.md [hunk 12]
```diff
@@ -225,11 +272,23 @@ function a() {
   console.log(x, y, z); // 3 4 5
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/statements/var/index.md:272]: 268行的括号

**Review by yin1999 — APPROVED** [review]
Thank you @JasonLamv-t

**Pull request merged**